### DISCUSSION TITLE
How to use Web technologies with Avalonia for Mobile/Desktop apps?

### DISCUSSION THREAD

**Discussion by alessandrofrancesconi** [AvaloniaUI/Avalonia#16189] [Q&A]
Hi everyone, I'm using Avalonia since 0.10.x version and I'm quite satisfied about it. So far I used it for making both Desktop and Android/iOS apps, writing in plain XAML/C# without particular extensions.

But I'm not an expert on XAML and, so far, my apps had a very basic (_ugly!_) UI. Instead, I'm quite much skilled on making web-based interfaces. I know how to use Javascript libraries to make animations, draw interactive chars, 3D graphics and so on.

So my question is, how can I exploit my skills on web frontends to make a visually cool Avalonia app, for Mobile and Desktop platforms? Something that lets me write the app in HTML/JS/CSS for the frontend and C# for the logical part.

Are there some Avalonia "extensions" for this? And side questions...

- If such framework exists, does it let me use any existing Javascript library (including 3D plotting libraries like [Plotly](https://plotly.com/javascript/))?
- Does it let me keep the compatibility with Android, iOS and also **Windows 7**? I still have quite old costumers...

Thanks!

**Comment by timunie** [comment]
I don't think there's an easy way to work with HTML and JS in Avalonia. These are different technologies. Even if there was a good enough web view, I think that you should better try to use XAML or use a different framework if Avalonia can't offer what you need.

**Reply by alessandrofrancesconi** [reply]
Thanks, but what about Avalonia+Blazor? Could it be a way to develop applications with a web-based frontend, for Desktop and Mobile (not considering browsers)?

**Reply by maxkatz6** [reply]
> Avalonia+Blazor

While possible in theory, it has little benefits. As point of Avalonia is to have windowing/rendering/input/layout to be done without browser. If you bring browser into to the equation, you might as well just use existing simpler browser wrappers, like:
- https://github.com/ElectronNET/Electron.NET
- https://learn.microsoft.com/en-us/aspnet/core/blazor/hybrid/?view=aspnetcore-8.0

**Comment by tkefauver** [comment]
[Avalonia.Webview](https://github.com/MicroSugarDeveloperOrg/Avalonia.WebView) has worked well for me and they have Blazor samples but like @maxkatz6 said there's probably not much benefit if you don't use Avalonia in the front-end.

**Reply by lostmsu** [reply]
On my first attempt integrating it into my app it simply showed nothing. The sample worked fine. The project had no commits in the last year. Avalonia desperately needs a built-in and supported WebView.
